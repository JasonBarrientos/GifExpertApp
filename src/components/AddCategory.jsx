import { useState } from 'react'
import PropTypes from "prop-types";
export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState('')
    const onInputChanged = ({ target }) => {
        setInputValue(target.value)
    }
    const onSubmit = (event) => {
        event.preventDefault(); //evita q se recargue todo
        const newInpuValue = inputValue.trim()
        if (newInpuValue.length <= 1) return;
        onNewCategory(newInpuValue)
        setInputValue('')
    }
    return (
        <form onSubmit={onSubmit} aria-label='form' >
            <input
                type="text"
                placeholder='Buscar gif'
                value={inputValue}
                onChange={(e => {
                    onInputChanged(e);
                })} />
        </form>

    )
}

AddCategory.propTypes={
    onNewCategory: PropTypes.func.isRequired
}