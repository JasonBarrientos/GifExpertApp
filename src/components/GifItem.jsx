import PropTypes from "prop-types";

export const GifItem = ({title, url}) => {    
  return (
    <div className='card'>
      <p>{title}</p>
      <img src={url} alt={title} />
    </div>
  )
}

GifItem.protTypes={
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}