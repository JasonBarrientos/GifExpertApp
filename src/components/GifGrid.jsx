import { useEffect, useState } from "react";
import { getGif } from "../helpers/getGifs";
import {GifItem} from './GifItem'

export const GifGrid = ({ category }) => {
    const [images, setImages] = useState([]);

    const getImages= async (category)=>{
        const newImages=  await  getGif(category)
        setImages(newImages)
    }

   useEffect(() => {
    getImages(category)
   }, [])
   
    return (
        <>
            <h3>{category}</h3>
            <div className="card-grid">
                {
                    images.map(image=>{
                      return  <GifItem key={image.id} {...image} ></GifItem>
                    })
                }
            </div>
        </>
    )
}

// GifGrid.prototype({
//     category: require()
// })