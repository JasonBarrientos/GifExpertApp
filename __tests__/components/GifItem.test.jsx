import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";


describe('Pruebas component <GifItem></GifItem> ', () => { 
const title= 'titulo prueba';
const url='http://url.prueba.com'

    test('Debe coincidir con el snatshop', () => { 

        const {container}= render(<GifItem title={ title  } url= { url}/>)
        expect(container).toMatchSnapshot()
     })
     test('Debe demsotrar la iamgen con el url y el alt indicado', () => { 
        render(<GifItem title={ title  } url= { url}/>)
        const {src,alt}=screen.getByRole('img')
        expect(src).toContain(url)
        expect(alt).toContain(title)
      })
       test('Debe demsotrar titulo en el componente', () => { 
        render(<GifItem title={ title  } url= { url}/>)
        expect(screen.getByText(title)).toBeTruthy();
      })

 })