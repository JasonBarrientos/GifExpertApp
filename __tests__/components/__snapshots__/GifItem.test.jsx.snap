// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas component <GifItem></GifItem>  Debe coincidir con el snatshop 1`] = `
<div>
  <div
    class="card"
  >
    <p>
      titulo prueba
    </p>
    <img
      alt="titulo prueba"
      src="http://url.prueba.com"
    />
  </div>
</div>
`;
